src/lib/transition_frontier/persistent_root/persistent_root.ml

```diff
@@ -2,13 +2,13 @@ open Core
 open Mina_base
 module Ledger = Mina_ledger.Ledger
 open Frontier_base
-module Ledger_transfer = Mina_ledger.Ledger_transfer.Make (Ledger) (Ledger.Db)
-module Ledger_transfer_stable =
-  Mina_ledger.Ledger_transfer.Make (Ledger.Db) (Ledger.Db)
+module Ledger_transfer_any =
+  Mina_ledger.Ledger_transfer.Make (Ledger.Any_ledger.M) (Ledger.Any_ledger.M)
 
-let transfer_snarked_root =
-  Ledger.Root.transfer_accounts_with
-    ~stable:Ledger_transfer_stable.transfer_accounts
+let transfer_snarked_root ~src ~dest =
+  Ledger_transfer_any.transfer_accounts
```

Comment:
I kept the use of `Ledger_transfer_any.transfer_accounts` in this method. It's used exactly once, when populating the root snarked ledger from a recovered potential snarked ledger.

This could be removed too, if reading all the accounts in the database and regenerating the database structure is not important for this recovery mechanism.